src/main/java/org/kohsuke/github/GHContentUpdateResponse.java

```diff
@@ -21,7 +25,101 @@ public GHContent getContent() {
      *
      * @return the commit
      */
-    public GHCommit getCommit() {
+    public Commit getCommit() {
         return commit;
     }
+
+    /**
+     * The type Commit.
+     */
+    public static class Commit {
+        String sha;
+        GitUser author;
+        GitUser committer;
+        String message;
+        Tree tree;
+        String url;
+
+        /**
+         * Gets sha.
+         *
+         * @return the sha
+         */
+        public String getSha() {
+            return this.sha;
+        }
```

Comment:
This is not committable, but this is what I'm thinking given our discussion so far. 
```suggestion
    public static class Commit extends GHCommit.ShortInfo {
        String sha;

        /**
         * Gets sha.
         *
         * @return the sha
         */
        public String getSha() {
            return this.sha;
        }
        
        
        /**
         * Gets comment count.
         *
         * @return the comment count
         */
        @Override
        public int getCommentCount() {
            throw GHException("Not available on this endpoint.");
        }
```

And then you can delete the rest.

Reply:
Hmm... `tree` is not accessible in `GHCommit.ShortInfo`.

Reply:
Hmm... `tree` in `GHCommit.ShortInfo` is not accessible from a subclass. I think reusing `GHCommit.ShortInfo` may not be a good idea.

Reply:
@takezoe 
It is okay to change internals or to add methods if you believe they make sense.  

You could change `tree` to be `protected` so that it could be accessed here. You'll note though that `ShortInfo.Tree` is not `public`.  Instead, `GHCommit` uses it to request a `GHTree` if the user asks for it.  

I should say here that you've repeated indicated that you don't believe extending like this is the right path.  I'm open to alternatives, but I also don't want to just pile more classes on because GitHub didn't make their API consistent. 

Maybe there should be a common base class for these two classes?  Maybe a `GitCommit` similar to `GitUser`?

Reply:
> You could change tree to be protected so that it could be accessed here. You'll note though that ShortInfo.Tree is not public. Instead, GHCommit uses it to request a GHTree if the user asks for it.

`ShortInfo.Tree` has only `sha` because it seems to be designed to use to create `GHCommit` by an extra API call when asked. But in some other classes, there are `Tree` that have `sha` and `url` and be accessible from users directly.

https://github.com/hub4j/github-api/blob/cf2571858c429967764fdb4e9cd38aae6a18bbe1/src/main/java/org/kohsuke/github/GHCompare.java#L250-L251

https://github.com/hub4j/github-api/blob/cf2571858c429967764fdb4e9cd38aae6a18bbe1/src/main/java/org/kohsuke/github/GHPullRequestCommitDetail.java#L59-L61

I don't know which design is better. If we only need `tree.sha` or `tree.url`, we don't need to make such an extra API call to create `GHTree`. Personally, I prefer this because it allows us to avoid an implicit extra API call.

> I also don't want to just pile more classes

Hmm... If that's a philosophy of this library, why there are duplicated `Tree`s in some classes? Why there are classes that return `Tree` and `GHTree`? It's hard to find the consistency in the existing codebase of this library and what you are saying.

> because GitHub didn't make their API consistent.

Does this mean GitHub modifies API frequently so you want to minimize necessary future effort to follow the changes? Yeah, I totally agree with it!

> Maybe there should be a common base class for these two classes? Maybe a GitCommit similar to GitUser?

On the other hand, looking at use of bridge methods, this library seems to emphasize the backward compatibility, meaning we cannot delete unnecessary properties easily once added. In that case, having a base class could increase the chance of sparse properties (like `commit_count`).

Reply:
To be honest, I cannot find what is the best way to fix this issue from the existing codebase and what you are saying probably because of the lack of my understanding and knowledge about this library. So, can I ask you to fix this issue in a way you want? Obviously, this issue must be fixed, but I don't want to spend more time for this.

Reply:
@takezoe 
It is unfortunate that you have no more time.  This project is many years old and has grown organically along with the GitHub REST API.  I understand your frustration.  You figured you'd just drop in fix an obviously broken feature and I've made it much more involved.   Sorry about that.

I agree that all the things you mention above are real problems and I sympathize with your frustration. However, all the problems you mention only highlight the need for cleanup and improvement in this area.  After a bit of digging in the docs, I found an endpoint that [gets a Git Commit](https://docs.github.com/en/rest/reference/git#get-a-commit). So, these various other classes should probably all be inheriting from a single base called `GitCommit` which should be returned by `GHRepository.getGitCommit()`.

I guess I'll file an issue that points the this PR and hopefully someone will have time to fix this eventually.

Reply:
No worries. You a maintainer of this library. Your thought should be respected. I just don't have a time to do try-and-error to find the best way for this.

> However, all the problems you mention only highlight the need for cleanup and improvement in this area.

That may be true. For `GHCommit`, I think it should provide a way to get raw `tree` information without an extra API call. Why we need to make an extra API call when we only need `tree.sha`?

Reply:
> > However, all the problems you mention only highlight the need for cleanup and improvement in this area.
> 
> That may be true. For `GHCommit`, I think it should provide a way to get raw `tree` information without an extra API call. Why we need to make an extra API call when we only need `tree.sha`?

Less than optimal design.  There should definitely be a way to do what you are pointing to.  Please feel free to file issues/PRs to address this and any other of the points you mentioned above.